tests/Tests.AzureAppConfiguration/PushRefreshTests.cs

```diff
@@ -0,0 +1,333 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Azure;
+using Azure.Core.Testing;
+using Azure.Data.AppConfiguration;
+using Azure.Messaging.EventGrid;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Xunit;
+
+namespace Tests.AzureAppConfiguration
+{
+	public class PushRefreshTests
+	{
+		List<ConfigurationSetting> _kvCollection = new List<ConfigurationSetting>
+		{
+			ConfigurationModelFactory.ConfigurationSetting(
+				key: "TestKey1",
+				label: "label",
+				value: "TestValue1",
+				eTag: new ETag("0a76e3d7-7ec1-4e37-883c-9ea6d0d89e63"),
+				contentType: "text"),
+
+			ConfigurationModelFactory.ConfigurationSetting(
+				key: "TestKey2",
+				label: "label",
+				value: "TestValue2",
+				eTag: new ETag("31c38369-831f-4bf1-b9ad-79db56c8b989"),
+				contentType: "text"),
+
+			ConfigurationModelFactory.ConfigurationSetting(
+				key: "TestKey3",
+				label: "label",
+				value: "TestValue3",
+				eTag: new ETag("bb203f2b-c113-44fc-995d-b933c2143339"),
+				contentType: "text"),
+
+			ConfigurationModelFactory.ConfigurationSetting(
+				key: "TestKeyWithMultipleLabels",
+				label: "label1",
+				value: "TestValueForLabel1",
+				eTag: new ETag("bb203f2b-c113-44fc-995d-b933c2143339"),
+				contentType: "text"),
+
+			ConfigurationModelFactory.ConfigurationSetting(
+				key: "TestKeyWithMultipleLabels",
+				label: "label2",
+				value: "TestValueForLabel2",
+				eTag: new ETag("bb203f2b-c113-44fc-995d-b933c2143339"),
+				contentType: "text")
+		};
+
+        List<PushNotification> pushNotificationList = new List<PushNotification>
```

Comment:
>pushNotificationList [](http://example.com/codeflow?start=31&length=20)

We should use [_camelCase](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions#camel-case) naming for all private fields.